src/SigasiProjectCreator/ArgsAndFileParser.py

```diff
@@ -3,61 +3,223 @@
+                                 dest='uvmlib', default=None)
+        self.parser.add_argument('--format', action='store', dest='format',
+                                 choices=['dotf', 'csv', 'filelist', 'hdp', 'xise'], default=None,
+                                 help='Force input format (ignore file extension)')
+        self.parser.add_argument('--mapping', action='store', dest='mapping',
+                                 choices=['file', 'folder'], default='file',
+                                 help='Library mapping style: `folder` = map folders where possible, `file` = map '
+                                      'individual files (default). Option `folder` requires that files are actually '
+                                      'available. Only relevant with `default`, `linked-files-tree` and '
+                                      '`linked-folders` project layouts')
+        self.parser.add_argument('--enable-vhdl', action='store_true', dest='enable_vhdl',
+                                 help='Force VHDL support (regardless of VHDL file presence)')
+        self.parser.add_argument('--vhdl-version', action='store', dest='vhdl_version',
+                                 choices=VhdlVersion.get_str_enums(), default=str(VhdlVersion.TWENTY_O_EIGHT),
+                                 help='Set VHDL version (default VHDL-2008)')
+        self.parser.add_argument('--enable-verilog', action='store_true', dest='enable_verilog',
+                                 help='Force (System)Verilog support (regardless of (System)Verilog file presence)')
+        self.parser.add_argument('--verilog-as-sv', action='store_true', dest='system_verilog',
+                                 help='Treat .v files as SystemVerilog')
+        self.parser.add_argument('--enable-vunit', action='store_true', dest='enable_vunit',
+                                 help='Enable VUnit support')
+        self.parser.add_argument('-w', '--work', help='Main HDL library name (default `work`)',
+                                 dest='worklib', default='work')
+        self.parser.add_argument('--skip-check-exists', action='store_true', dest='skip_check_exists',
+                                 help='Skip checking whether files and folders exist')
+        self.parser.add_argument('--encoding', action='store', dest='encoding',
+                                 default='UTF-8', help='Set unicode character encoding (default: UTF-8)')
+        self.parser.add_argument('-f', '--force', action='store_true', dest='force_write',
+                                 help='Overwrite existing project files')
+        self.parser.add_argument('--rel-path', action='store', dest='rel_path_root',
+                                 nargs='*', type=pathlib.Path,
+                                 help='Use relative paths for links to files in this folder and its sub-folders')
+
+    @staticmethod
+    def get_file_type(filename):
```

Comment:
Static methods are rarely used in python; normally you'd put this as a toplevel function in `ArgsAndFileParser`, and use it like this:

```python
import SigasiProjectCreator.ArgsAndFileParser

ArgsAndFileParser.get_file_type(...)
```

Reply:
static methods all cleaned up